src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -52,29 +67,118 @@ public DefaultAblyBatchClient(ChannelSinkConnectorConfig connectorConfig, Channe
+
+                // Step 2: Parse the response.
+                // If its 4xx or 5xx, then DLQ all the sink records.
+                // If its 2xx, then DLQ only the failed sink records for that specific channel.
+                ABLY_BATCH_RESPONSE ablyBatchResponse = parseAblyBatchAPIResponse(response);
+
+                if(ablyBatchResponse == ABLY_BATCH_RESPONSE.FAILURE) {
+                    sendMessagesToDLQ(records, dlqReporter);
+                } else if(ablyBatchResponse == ABLY_BATCH_RESPONSE.PARTIAL_FAILURE) {
+                    handlePartialFailure(response, channelNameToSinkRecordsMap, dlqReporter);
+                }
+
             } catch (Exception e) {
                 logger.error("Error while sending batch", e);
+                sendMessagesToDLQ(records, dlqReporter);
+            }
+        }
+    }
+
+    /**
+     * Function to handle partial failure
+     * @param response Ably response
+     * @param records  Sink records
+     * @param dlqReporter DLQ reporter
+     */
+    public void handlePartialFailure(HttpPaginatedResponse response,
+                                     Map<String, List<SinkRecord>> records,
+                                     ErrantRecordReporter dlqReporter) {
+
+        Set<String> failedChannels = new HashSet<>();
+        JsonElement[] elements = response.items();
+        for (JsonElement element : elements) {
+            failedChannels.addAll(getFailedChannels(element));
+        }
+
+        for(String channel : failedChannels) {
+            List<SinkRecord> failedRecords = records.get(channel);
+            if(failedRecords != null || !failedRecords.isEmpty()) {
+                sendMessagesToDLQ(failedRecords, dlqReporter);
+            }
+        }
+    }
+
+    /**
+     * Function to parse Ably Batch API response
+     * @param response Ably response
+     * @return ABLY_BATCH_RESPONSE
+     */
+    public ABLY_BATCH_RESPONSE parseAblyBatchAPIResponse(HttpPaginatedResponse response) {
+
+        ABLY_BATCH_RESPONSE ablyBatchResponse = ABLY_BATCH_RESPONSE.SUCCESS;
+
+        String statusCode = Integer.toString(response.statusCode);
+        if(statusCode.startsWith(ABLY_REST_API_ERROR_CODE_4XX) || statusCode.startsWith(ABLY_REST_API_ERROR_CODE_5XX)) {
+            // DLQ all the sink records.
+            logger.error("Ably Batch API call failed with error code: " + statusCode);
+            ablyBatchResponse = ABLY_BATCH_RESPONSE.FAILURE;
+        } else if(statusCode.startsWith(ABLY_REST_API_ERROR_CODE_2XX)) {
+            JsonElement[] elements = response.items();
+            for (JsonElement element : elements) {
+                int failureCount = element.getAsJsonObject().getAsJsonPrimitive("failureCount").getAsInt();
+                if (failureCount > 0) {
+                    // DLQ only the failed sink records for that specific channel.
+                    logger.error("Ably Batch API call failed with error code: " + statusCode);
```

Comment:
Might be useful to distinguish this log line from the response-level error logging above?